src/views/profile-view/ProfileAddLink.vue

```diff
@@ -0,0 +1,252 @@
+        },
+        {
+          check: (value: string) => value.length < 3,
+          message: "Should be 3 or more characters",
+        },
+      ],
+    });
+
+    // TODO: update validation rules for link
+    const {
+      value: link,
+      error: linkError,
+      errorMessage: linkErrorMessage,
+      isValid: linkIsValid,
+      validate: validateLink,
+    } = useValidation({
+      initialValue: "",
+      rules: [
+        {
+          check: (value: string) => (value ? false : true),
+          message: "link is required",
+        },
+        {
+          check: (value: string) => value.length < 3,
+          message: "Should be 3 or more characters",
+        },
+      ],
+    });
+
+    return {
+      linkType,
+      title,
+      titleError,
+      titleErrorMessage,
+      titleIsValid,
+      validateTitle,
+      description,
+      descriptionError,
+      descriptionErrorMessage,
+      descriptionIsValid,
+      validateDescription,
+      link,
+      linkError,
+      linkErrorMessage,
+      linkIsValid,
+      validateLink,
+      step,
+      isAddLink
+    }
+  },
+  computed: {
+    canAddLink(): boolean {
+      return this.titleIsValid && this.descriptionIsValid && this.linkIsValid;
+    },
+  },
+  methods: {
+    async addLink() {
+      this.isAddLink = true;
+      const dataStore = useDataStore();
+      const userStore = useUserStore();
+      const appStore = useAppStore();
+      const userPerspective = userStore.getFluxPerspectiveId;
+
+      // @ts-ignore
+      const { links: preLinks } = await ad4mClient.perspective.snapshotByUUID(
+        userPerspective!
+      );
+
+      const preArea: {[x: string]: any} = {};
+
+      preLinks.forEach((e: any) => {
+        const predicate = e.data.predicate.split('://')[1];
+        if (!preArea[e.data.source]) {
+          preArea[e.data.source] = {
+            [predicate]: predicate === 'has_post' ? e.data.target : e.data.predicate.split('://')[1],
+          }
+        }
+
+        preArea[e.data.source][predicate] = e.data.predicate.split('://')[1];
+      });
+      console.log('preLinks', preLinks)
+      await ad4mClient.perspective.addLink(
+        userPerspective!,
+        new Link({ source: `area-${Object.keys(preArea).length}`, target: this.link, predicate: "sioc://has_post" })
+      );
+      await ad4mClient.perspective.addLink(
+        userPerspective!,
+        new Link({ source: `area-${Object.keys(preArea).length}`, target: 'flux://community', predicate: "flux://area_type" })
+      );
+      await ad4mClient.perspective.addLink(
+        userPerspective!,
+        new Link({ source: `area-${Object.keys(preArea).length}`, target: `text://${this.title}`, predicate: "sioc://has_name" })
+      );
+      await ad4mClient.perspective.addLink(
+        userPerspective!,
+        new Link({ source: `area-${Object.keys(preArea).length}`, target: `text://${this.description}`, predicate: "sioc://has_description" })
+      );
```

Comment:
I wonder if there is a higher level abstraction / pattern we can create to avoid most manual link traversing / filtering, might be worth us having a discussion together with @leifriksheim

Reply:
Would it be good to store it in an expression? Because Lief and I were discussing that what if we are storing metadata for web links we could end up having 10+ links for a single area.

Reply:
Yeah I think we will get to that point where we have to start storing the areas as expressions, we can definitely look into that once we finish this first implementation